File: app/Models/Zipcode.php
<?php

namespace App\Models;

use Laravel\Sanctum\HasApiTokens;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Notifications\Notifiable;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Zipcode extends Model
{
    use HasApiTokens, HasFactory, Notifiable;
    protected $fillable = [
        'zipcode',
        'city_id',
        'minimum_free_delivery_order_amount',
        'delivery_charges',
    ];

    public function cities()
    {
        return $this->hasMany(City::class);
    }
    public function city()
    {
        return $this->belongsTo(City::class);
    }
}

File: customer_app/lib/ui/profile/address/models/zipcode.dart
class Zipcode {
  int? id;
  String? zipcode;
  int? cityId;
  int? minimumFreeDeliveryOrderAmount;
  int? deliveryCharges;
  String? createdAt;
  String? updatedAt;

  Zipcode(
      {this.id,
      this.zipcode,
      this.cityId,
      this.minimumFreeDeliveryOrderAmount,
      this.deliveryCharges,
      this.createdAt,
      this.updatedAt});

  Zipcode.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    zipcode = json['zipcode'];
    cityId = json['city_id'];
    minimumFreeDeliveryOrderAmount = json['minimum_free_delivery_order_amount'];
    deliveryCharges = json['delivery_charges'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
  }



}
